### PR TITLE
Accessible names for template bulk selection checkboxes

### PR DESCRIPTION
**Pull request by amazingphilippe**
# Summary | Résumé

Fixes https://github.com/cds-snc/notification-planning/issues/60

# Test instructions | Instructions pour tester la modification

1. On the template directory page. Turn on your screen reader. 
2. Using TAB, navigate through the list.
3. The checkbox should announce what it folder, sms template or email template it belongs to.

# Reviewer checklist | Liste de vérification du réviseur

This is a suggested checklist of questions reviewers might ask during their
review | Voici une suggestion de liste de vérification comprenant des questions
que les réviseurs pourraient poser pendant leur examen :

- [ ] Does this meet a user need? | Est-ce que ça répond à un besoin utilisateur?
- [ ] Is it accessible? | Est-ce que c’est accessible?
- [ ] Is it translated between both offical languages? | Est-ce dans les deux
      langues officielles?
- [ ] Is the code maintainable? | Est-ce que le code peut être maintenu?
- [ ] Have you tested it? | L’avez-vous testé?
- [ ] Are there automated tests? | Y a-t-il des tests automatisés?
- [ ] Does this cause automated test coverage to drop? | Est-ce que ça entraîne
      une baisse de la quantité de code couvert par les tests automatisés?
- [ ] Does this break existing functionality? | Est-ce que ça brise une
      fonctionnalité existante?
- [ ] Should this be split into smaller PRs to decrease change risk? | Est-ce
      que ça devrait être divisé en de plus petites demandes de tirage (« pull
      requests ») afin de réduire le risque lié aux modifications?
- [ ] Does this change the privacy policy? | Est-ce que ça entraîne une
      modification de la politique de confidentialité?
- [ ] Does this introduce any security concerns? | Est-ce que ça introduit des
      préoccupations liées à la sécurité?
- [ ] Does this significantly alter performance? | Est-ce que ça modifie de
      façon importante la performance?
- [ ] What is the risk level of using added dependencies? | Quel est le degré de
      risque d’utiliser des dépendances ajoutées?
- [ ] Should any documentation be updated as a result of this? (i.e. README
      setup, etc.) | Faudra-t-il mettre à jour la documentation à la suite de ce
      changement (fichier README, etc.)?

### CODE REVIEW

File: app/templates/components/checkbox.html [hunk 1]
```diff
@@ -38,9 +38,11 @@
...
 {% endmacro %}
 
-{% macro unlabelled_checkbox(id, name, data=None, value="y") %}
+{% macro unlabelled_checkbox(id, name, data=None, value="y", label="") %}
   <div class="multiple-choice">
     {{ checkbox_input(id, name, data, value) }}
-    <label></label>
+    <label for="{{id}}">
+      <span class="sr-only">{{label}}</span>
+    </label>
```
Comment (jimleroyer) [inline app/templates/components/checkbox.html:46]: Are we not duplicating the text that is sitting beside folder icon? I would expect a `label` element sitting there normally. Is my thinking wrong?

**Review by jimleroyer — COMMENTED** [review]
There is a failing unit test following this change. Tell me if you need some dev help on that!

File: app/templates/components/checkbox.html (hunk 1 above)
Comment (amazingphilippe) [reply app/templates/components/checkbox.html:46]: Not all wrong. The template list is a little funky. But this kinda makes sense. 

1. The checkbox is for bulk select.
2. The list item is a link that navigates to the template

I think the idea is to separate the two interactions. I might be wrong. There could be another solution to the missing label...

Comment (amazingphilippe) [reply app/templates/components/checkbox.html:46]: @jimleroyer I have another idea on how we can fix this by using aria-labelledby, but we lose the surrounding words that would help you know why there is a checkbox

Comment (jimleroyer) [reply app/templates/components/checkbox.html:46]: Let's shoot for what you think is feasible. 👍 I am fine either way!

Comment (amazingphilippe) [reply app/templates/components/checkbox.html:46]: Grouped list in a fieldset, but kept the hidden labels after considering that the impact of checkbox select is better described with custom label text. Thanks for the review!

**Review by jimleroyer — APPROVED** [review]
:shipit:

**Pull request merged**